doc/glossary.rst

```diff
@@ -0,0 +1,165 @@
+NeuroDSP Glossary
+=================
+
+The following is a glossary of neuroscience and digital processing related terms that are used in NeuroDSP.
+
+General
+-------
+
+.. glossary::
+
+    periodic
+        Properties or components of a signal that are rhythmic.
+    aperiodic
+        Properties or components of a signal that are arrhythmic, with no characteristic frequency.
+
+Digital Signal Processing
+-------------------------
+
+For a general introduction to digital signal processing, we recommend
+`Seeing Circles Sines and Signal <https://jackschaedler.github.io/circles-sines-signals/>`_
+by Jack Schaedler.
+
+.. glossary::
+
+    time domain
+        Signals that are represented as variations over time, and analyses of such signals.
+    frequency domain
+        Signals that are represented in terms of frequencies, and Analyses of such signals.
+    temporal resolution
+        The precision of a measurement, in the time domain.
+    frequency resolution
+        The precision of a measurement, in the frequency domain.
+    sampling rate
+        The rate at which samples are taken.
+
+Units
+-----
+
+.. glossary::
+
+    Hertz (Hz)
+        A unit of frequency, as the number of cycles per second.
+    Decibels (dB)
+        A unit of intensity, on a logarithmic scale.
+    Volts (V)
+        A unit of voltage, typically in the microvolt (uV) range for neural time series.
+
+Filters
+-------
+
+For a guide on filtering, specific to electrophysiological data, check out this
+`paper <https://doi.org/10.1016/j.jneumeth.2014.08.002>`_ from the journal of neuroscience methods.
+
+For a more in depth tutorial, in code, check out the
+`MNE Filtering Tutorial <https://martinos.org/mne/stable/auto_tutorials/plot_background_filtering.html>`_.
+
+.. glossary::
+
+    Impulse Response
+        The response of a fitler when presented with an impulse; a single, brief input.
+    FIR
+        A Finite Impulse Response filter, meaning the response to a single input is finite, settling to zero.
```

Comment:
"...meaning its impulse response settles to zero in finite time."

Reply:
I just added impulse responses cuz I think it helps linking it with the concept above.